### ISSUE TITLE
Cluster Replication

### ISSUE THREAD

**Issue by jwmarshall** [neildunbar/mariadb55#3]
I discovered a bug where the replication client on the primary node could not sync the database to a new cluster node. The problem ultimately lead me to a permissions issue with the 'mysql' user on the primary node. The following grants need to be run in order to allow replication to function.

`GRANT RELOAD, SUPER, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'mysql'@'localhost'`

I tried forking your repository and adding the grants to the script that creates this user and sets the root password, however it did not function as expected. Either there are problems with your scripts and a newer version of mariadb, or perhaps there are some changes you made to that were never pushed to github.

Following your instructions in the README, but performing the grants listed above before adding additional cluster nodes, works perfectly.

**Comment by neildunbar** [comment]
good catch. I've been busy over the last few months, but I need to get this tidied up quite a bit.

**Comment by jwmarshall** [comment]
Looks like the following fork is ahead of you on a few other fixes as well. I'm testing it now, they fixed the permissions issue by granting all permissions to the 'mysql' user which is kind of like killing a fly with a shotgun. ;)

Anyway check out the changes made in this fork and consider merging them with your repository. I will let you know if my testing checks out.

https://github.com/nsasikumar/mariadb55
